vital/algo/resection_camera.cxx

```diff
@@ -0,0 +1,85 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief resection_camera instantiation
+
+#include <vital/algo/algorithm.txx>
+#include <vital/algo/resection_camera.h>
+
+/// \cond DoxygenSuppress
+INSTANTIATE_ALGORITHM_DEF( kwiver::vital::algo::resection_camera );
+/// \endcond
+
+using namespace std;
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+// ----------------------------------------------------------------------------
+resection_camera
+::resection_camera()
+{
+  attach_logger( "algo.resection_camera" );
+}
+
+// ----------------------------------------------------------------------------
+camera_perspective_sptr
+resection_camera
+::resection( frame_id_t frame_id,
+             landmark_map_sptr landmarks,
+             feature_track_set_sptr tracks,
+             unsigned width, unsigned height ) const
+{
+  // Generate calibration guess from image dimensions
+  auto const principal_point = vector_2d{ width * 0.5, height * 0.5 };
+  auto cal = std::make_shared< simple_camera_intrinsics >(
+    ( width + height ) * 0.5, principal_point, 1.0, 0.0,
+    Eigen::VectorXd(), width, height );
+
+  // Resection using guessed calibration
+  return resection( frame_id, landmarks, tracks, cal );
+}
+
+// ----------------------------------------------------------------------------
+camera_perspective_sptr
+resection_camera
+::resection( frame_id_t frame_id,
+             landmark_map_sptr landmarks,
+             feature_track_set_sptr tracks,
+             kwiver::vital::camera_intrinsics_sptr cal ) const
+{
+  auto world_points = vector< vector_3d >{};
+  auto camera_points = vector< vector_2d >{};
+
+  auto const& real_landmarks = landmarks->landmarks();
+
+  for( auto const& track : tracks->tracks() )
```

Comment:
`feature_track_set` has a helper to make this more efficient. `tracks->frame_feature_track_states(frame_id)`

```c++
for( auto const& fts : tracks->frame_feature_track_states(frame_id) )
{
  auto lmi = real_landmarks.find( fts->track()->id() );
  ...
```

Reply:
This is not critical, just an optimization.  We can revise this later if it would help get this branch landed sooner.

Reply:
done